src/pal/pal_noalloc.h

```diff
@@ -17,17 +17,18 @@ namespace snmalloc
    * address-space manager is initialised with all of the memory that it will
    * ever use.
    *
-   * It takes an error handler delegate as a template argument. This is
+   * It takes an underlying PAL delegate as a template argument. This is
    * expected to forward to the default PAL in most cases.
    */
-  template<typename ErrorHandler>
+  template<SNMALLOC_CONCEPT(ConceptPAL) UnderlyingPAL>
```

Comment:
Please don't make this require the PAL concept.  This does not need to implement all of the features of a PAL, it needs only to provide error handling code.  In my existing use, the error handler is significantly simpler than a full PAL.  In particular, it can't allocate or deallocate memory.

Reply:
I agree that that I should just go back to a typename, but for the OpenEnclave platform it is not just the error handling that needs delegation, but also the zeroing interface and the entropy generation. The latter is also not needed if the UnderlyingPAL does not specify Entropy as an available feature.

Reply:
I like the idea of having other PALs replace this one and not fit in as template arguments, but then we need to add a PAL template to FixedGlobals which defaults to Pal. Does that sound good @mjp41 ?